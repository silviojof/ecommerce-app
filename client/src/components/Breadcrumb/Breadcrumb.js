import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import styles from './Breadcrumb.module.scss';

export const Breadcrumb = ({ categories }) => {
    return (
        <div className={styles.breadcrumb}>
            {
                categories.map((el, index) => {
                    const isLast = index !== categories.length - 1;
                    return (
                        <div className={styles.breadcrumbItem} key={el}>
                            <span className={`${isLast ? '' : styles.last}`}>{el}</span>
                            {
                                isLast &&
                                <span className={styles.breadcrumbArrow}>></span>
                            }
                        </div>
                )})
            }
        </div>
    )
};

Breadcrumb.propTypes = {
    categories: PropTypes.arrayOf((PropTypes.string)).isRequired,
};

const mapStateToProps = (state) => {
    return {
        categories: state.productList.categories
    }
}

export default connect(
    mapStateToProps,
)(Breadcrumb);