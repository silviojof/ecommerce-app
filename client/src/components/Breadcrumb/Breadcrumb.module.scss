@import '../../styles/variables';
@import '../../styles/mediaQuery';

.breadcrumb {
    font-size: 14px;
    color: $mediumGrey;
    font-weight: 400;
    margin: $spacing1;
    @include lg {
        margin: $spacing1 0;
    }
    .breadcrumbItem {
        display: inline-block;
        .last {
            font-weight: 600;
        }
    }
    .breadcrumbArrow {
        padding: 0 10px;
    }
}