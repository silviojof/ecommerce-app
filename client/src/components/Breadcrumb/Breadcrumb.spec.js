import React from 'react';
import {render} from 'react-testing-library';
import { Breadcrumb } from './Breadcrumb';

it ('matches snapshot', () => {
  const { container } = render(<Breadcrumb categories={['first', 'second']} />);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders two categories', () => {
  const {getByText} = render(<Breadcrumb categories={['first', 'second']} />);
  expect(getByText('first')).toBeInTheDocument();
  expect(getByText('first')).toBeInTheDocument();
});

it('renders last one with "last" class', () => {
    const {queryAllByText} = render(<Breadcrumb categories={['first', 'second']} />);
    expect(queryAllByText('>').length).toBe(1);
});