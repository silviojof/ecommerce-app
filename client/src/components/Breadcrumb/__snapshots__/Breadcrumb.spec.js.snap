// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="breadcrumb"
>
  <div
    class="breadcrumbItem"
  >
    <span
      class=""
    >
      first
    </span>
    <span
      class="breadcrumbArrow"
    >
      &gt;
    </span>
  </div>
  <div
    class="breadcrumbItem"
  >
    <span
      class="last"
    >
      second
    </span>
  </div>
</div>
`;
